core/tests/protractor_utils/BlogDashboardPage.js

```diff
@@ -0,0 +1,319 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Page object for the blog Dashboard page, for use
+ * in Protractor tests.
+ */
+
+var action = require('./action.js');
+var waitFor = require('./waitFor.js');
+var forms = require('../protractor_utils/forms.js');
+var workflow = require('../protractor_utils/workflow.js');
+var general = require('../protractor_utils/general.js');
+
+var BlogDashboardPage = function() {
+  var currUserProfilePhoto = element(
+    by.css('.protractor-test-profile-current-user-photo'));
+  var createBlogPostButton = element(
+    by.css('.protractor-test-create-blog-post-button'));
+  var editBlogPostButton = element(
+    by.css('.protractor-test-edit-blog-post-button'));
+  var unpublishBlogPostButton = element(
+    by.css('.protractor-test-unpublish-blog-post-button'));
+  var deleteBlogPostButton = element(
+    by.css('.protractor-test-delete-blog-post-button'));
+  var matTabLabels = element.all(by.css('.mat-tab-label'));
+  var draftTab = matTabLabels.get(0);
+  var publishTab = matTabLabels.get(1);
```

Comment:
IN continuation to above explanation, all the mat-tab-labels are picked up. We cannot use the name of the tab furhter to pick them up as the name of the tab changes according to the number of Blog posts it contains, thus making it difficult.
eg if 1 blog post is a draft it is 'DRAFTS (1)' if there will be 2 it will be 'DRAFTS (2)' and so on:
![Screenshot from 2021-08-18 11-04-14](https://user-images.githubusercontent.com/68547101/129842907-9863fa73-79ae-4fa2-925e-5d698c832481.png)

Reply:
What about something like this:

```text
for tab in matTabLabels {
  text = the text in the element tab
  if text starts with "DRAFTS" {
    ...
  } else {
    ...
  }
}

Reply:
Done